src/main/scala/chisel3/aop/Select.scala

```diff
@@ -512,27 +526,25 @@ object Select {
   // Collects when predicates as it searches through a module, then applying processCommand to non-when related commands
   private def searchWhens(module: BaseModule, processCommand: (Command, Seq[Predicate]) => Unit) = {
     check(module)
-    module._component.get.asInstanceOf[DefModule].commands.foldLeft((Seq.empty[Predicate], Option.empty[Predicate])) {
-      (blah, cmd) =>
-        (blah, cmd) match {
-          case ((preds, o), cmd) =>
-            cmd match {
-              case WhenBegin(_, Node(pred: Bool)) => (When(pred) +: preds, None)
-              case WhenBegin(_, l: LitArg) if l.num == BigInt(1) => (When(true.B) +: preds, None)
-              case WhenBegin(_, l: LitArg) if l.num == BigInt(0) => (When(false.B) +: preds, None)
-              case other: WhenBegin =>
-                sys.error(s"Something went horribly wrong! I was expecting ${other.pred} to be a lit or a bool!")
-              case _: WhenEnd => (preds.tail, Some(preds.head))
-              case AltBegin(_) if o.isDefined => (o.get.not +: preds, o)
-              case _: AltBegin =>
-                sys.error(s"Something went horribly wrong! I was expecting ${o} to be nonEmpty!")
-              case OtherwiseEnd(_, _) => (preds.tail, None)
-              case other =>
-                processCommand(cmd, preds)
-                (preds, o)
-            }
+
+    def searchCommands(
+      commands:       Seq[Command],
+      preds:          Seq[Predicate],
+      processCommand: (Command, Seq[Predicate]) => Unit
+    ): Unit = commands.foreach {
+      case FWhen(_, cond, ifRegion, elseRegion) =>
+        val pred = cond match {
+          case Node(pred: Bool) => When(pred)
+          case l: LitArg if l.num == BigInt(1) => When(true.B)
+          case l: LitArg if l.num == BigInt(0) => When(false.B)
+          case _ => ???
```

Comment:
```suggestion
          case _ =>
            sys.error(s"Something went horribly wrong! I was expecting $pred to be a lit or a bool!")
```
Any reason not to maintain the old error behavior?

Reply:
Not really. Fixed in c1f8b2302.